examples/SquareLine/WiFiClock/WiFiClock.ino

```diff
@@ -0,0 +1,980 @@
+    lv_obj_set_size(wifi_list, 300, 180);
+    for (int i = 0; i < num_wifi; i++) {
+        lv_obj_t *wifi_list_Item = lv_list_add_btn(wifi_list, NULL, WiFi.SSID(i).c_str());
+        lv_obj_set_user_data(wifi_list_Item, (void *)WiFi.SSID(i).c_str());
+        lv_obj_add_event_cb(wifi_list_Item, handleWifiListItemClick, LV_EVENT_ALL, NULL);
+    }
+    lv_obj_align(wifi_list, LV_ALIGN_CENTER, 0, 30);
+
+    lvgl_port_unlock();
+}
+
+void handlePasswordValidation()
+{
+    connected_count++;
+
+    if (WiFi.status() == WL_CONNECTED) {
+        Serial.println("password correct: Wifi connected success");
+
+        writeStrToNVS("NVS_WifiNa", selected_wifi_name);
+        writeStrToNVS("NVS_WifiPw", wifi_password);
+
+        wifi_connected_flag = true;
+        Serial.println("wifi_connected_flag: true");
+        writeIntToNVS("NVS_WifiFg", wifi_connected_flag);
+
+        password_valid_flag = false;
+        Serial.println("password_valid_flag: false");
+
+        connected_count = 0;
+
+        lvgl_port_lock(-1);
+        _ui_flag_modify(ui_SpinnerLoadPassword, LV_OBJ_FLAG_HIDDEN, _UI_MODIFY_FLAG_ADD);
+        _ui_flag_modify(ui_KeyboardPassword, LV_OBJ_FLAG_HIDDEN, _UI_MODIFY_FLAG_REMOVE);
+        _ui_screen_change(&ui_ScreenClock, LV_SCR_LOAD_ANIM_NONE, 0, 0, &ui_ScreenClock_screen_init);
+        lv_textarea_set_text(ui_TextPassword, "");
+        lvgl_port_unlock();
+
+        getWeather();
+    } else if(connected_count >= 6) {
+        Serial.println("password wrong: Wifi connected failed");
+
+        wifi_connected_flag = false;
+        Serial.println("password_valid_flag: false");
+        writeIntToNVS("NVS_WifiFg", wifi_connected_flag);
+
+        http.end();
+        WiFi.disconnect();
+
+        password_valid_flag = false;
+        Serial.println("wifi_connected_flag: false");
+
+        connected_count = 0;
+
+        lvgl_port_lock(-1);
+
+        _ui_flag_modify(ui_SpinnerLoadPassword, LV_OBJ_FLAG_HIDDEN, _UI_MODIFY_FLAG_ADD);
+        _ui_flag_modify(ui_KeyboardPassword, LV_OBJ_FLAG_HIDDEN, _UI_MODIFY_FLAG_REMOVE);
+        lv_textarea_set_text(ui_TextPassword, "");
+
+        lvgl_port_unlock();
+    }
+}
+
+void handleAlarmActivation()
+{
+    static const char * btns[] ={""};
+
+    lvgl_port_lock(-1);
+
+    lv_obj_t* Alarm_mbox = lv_msgbox_create(NULL, "Alarm activated", "Click the button to turn off the alarm", btns, true);
+    lv_obj_center(Alarm_mbox);
+
+    alarm_flag = false;
+    Serial.println("alarm_flag: false");
+
+    lvgl_port_unlock();
+}
+
+void scanAndDisplayWiFiList()
+{
+    if(wifi_list_switch) {
+        num_wifi = WiFi.scanNetworks();
+        Serial.println("Scan done");
+
+        if(num_wifi == 0) {
+            Serial.println("no networks found");
+        } else if(wifi_list_switch) {
+            wifi_list_switch = false;
+            Serial.println("wifi_list_switch: false");
+
+            displayWifiList();
+        }
+        WiFi.scanDelete();
+    }
```

Comment:
```suggestion
    if(!wifi_list_switch) {
        return;
    }
    
    num_wifi = WiFi.scanNetworks();
    Serial.println("Scan done");

    if(num_wifi == 0) {
        Serial.println("no networks found");
    } else if(wifi_list_switch) {
        wifi_list_switch = false;
        Serial.println("wifi_list_switch: false");

        displayWifiList();
    }
    WiFi.scanDelete();
```
How about this? And you can also modify other similar functions (like `onWeatherUpdate()` and `updateWeatherAndTime()`) in this way.